### PR TITLE
Convert IConfiguration use to Options pattern

### PR DESCRIPTION
**Pull request by rynowak**
This is better for refactoring (goodbye magic strings) and testability
(see TestOptions).

### CODE REVIEW

**Comment by MisterJames** [comment]
@BillWagner  I'm reviewing this one now.

**Comment by BillWagner** [comment]
@rynowak Can you address the merge conflicts, and we'll merge this in.

File: AllReadyApp/Web-App/AllReady/Services/SendGridDevelopmentWeb.cs [hunk 1]
```diff
@@ -21,44 +23,9 @@ public Task DeliverAsync(ISendGrid message)
...
             var resultObject = new object();
             client.DeliveryMethod = SmtpDeliveryMethod.SpecifiedPickupDirectory;
-            client.PickupDirectoryLocation = _config["DevelopmentEmailFolder"];
+            client.PickupDirectoryLocation = _settings.EmailFolder;
             client.SendAsync(new MailMessage(message.From.Address, message.To.FirstOrDefault().Address, message.Subject, message.Html), resultObject);
             return Task.FromResult(resultObject);
         }
     }
-
-    public static class ITransportAdapter
```
Comment (rynowak) [inline AllReadyApp/Web-App/AllReady/Services/SendGridDevelopmentWeb.cs]: This wasn't used anywhere - I had been under the assumption that @MisterJames had made changes that make this obsolete?

File: AllReadyApp/Web-App/AllReady/config.json [hunk 2]
```diff
@@ -18,13 +20,14 @@
...
-    "InsertSampleData": "true",
-    "InsertTestUsers": "true",
     "Storage": {
       "AzureStorage": "[storagekey]",    
       "EnableAzureQueueService": "false"
     }
   },
+  "Email": {
+    "EmailFolder": "..."
+  },
```
Comment (rynowak) [inline AllReadyApp/Web-App/AllReady/config.json]: This setting is used in the email sender but didn't have an entry in the config.

**Comment by rynowak** [comment]
@MisterJames @BillWagner  updated

**Pull request merged**